core/tests/ledger_cleanup.rs

```diff
@@ -621,8 +593,8 @@ mod tests {
         );
         let u1 = storage_previous;
 
-        // Poll on some compaction happening
-        info!("Begin polling for compaction ...");
+        // Poll on some cleanup happening
+        info!("Begin polling for cleanup ...");
         let start_poll = Instant::now();
         while blockstore.storage_size().unwrap_or(0) >= u1 {
             if start_poll.elapsed().as_secs() > ROCKSDB_FLUSH_GRACE_PERIOD_SECS {
```

Comment:
Per my other comment, we may end up spinning this whole duration if we don't push enough data.

Reply:
This check is no longer valid as it expects to see the storage size shink w/o further writes, and this relies on manual compaction to exactly purge the entries older than `max_ledger_shreds`. 

I think a more proper check would be checking whether the number of shreds older than `last_slot - DEFAULT_PURGE_SLOT_INTERVAL` is smaller than `max_ledger_shreds`, but hope this isn't something tricky as it depends on the background activity.

Reply:
Marking this one as resolved - will respond in the other spot where you had this comment.